controllers/cleanup.go

```diff
@@ -23,22 +24,27 @@ func (r *SearchReconciler) deleteClusterManagementAddon(instance *searchv1alpha1
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "search-collector",
 			Namespace: instance.GetNamespace(),
-			// Only delete the ClusterManagementAddon if it's owned by the Search operator.
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion: "search.open-cluster-management.io/v1alpha1",
-					Kind:       "Search",
-					Name:       "search-v2-operator",
-				},
-			},
 		},
 	}
-	err := r.Delete(context.TODO(), cma)
+	err := r.Get(context.TODO(), types.NamespacedName{Name: "search-collector",
+		Namespace: instance.GetNamespace()}, cma)
 	if err != nil && !errors.IsNotFound(err) {
-		log.Error(err, "Failed to delete ClusterManagementAddon", "name", cma)
+		log.Error(err, "Failed to get ClusterManagementAddon", "name", cma)
 		return err
 	}
-	log.Info("ClusterManagementAddon search-collector deleted", "name", cma)
+	for _, ref := range cma.ObjectMeta.OwnerReferences {
+		if ref.Kind == "Search" {
+			cma.OwnerReferences = nil
```

Comment:
I think only need to remove Search owner here because MCH also adds its owner to cma.